lib/array_equals.rb

```diff
@@ -1,5 +1,27 @@
 # Determines if the two input arrays have the same count of elements
 # and the same integer values in the same exact order
 def array_equals(array1, array2)
-  raise NotImplementedError
-end
+  if array1.nil? && array2.nil?
+    return true
+  
+  elsif array1.nil? || array2.nil?
+    return false
+  
+  elsif array1.length != array2.length
+    return false
+  
+  else
+    match_count = 0 
```

Comment:
Minor thing but you could remove the match_count variable and return False if array1[i] != array2[i] on line 16. As is, if the program execution reaches line 23, it would always return True since it would have exited early if a False condition was hit on 19, so you could remove the match_count check on that line.